### PR TITLE
Fix registrations padding; add registrations header; fix buttons

### LINKED ISSUE #128: Event detail view styling
### Describe the bug
- The event title is repeated, it is both in the toolbar and below the map image.
- When tapping the map preview there is no visual feedback for that tap, thus it is not directly clear it does something.
- There are 6 different types of fonts. This should be split in a title, subtitle for 'from', 'until', 'registration deadline' etc and regular text. Only a colour for emphasis on the registration terms could help there. Would be nice if these titles are re-used throughout the application, for example in settings for the 'notifications' text or the user profiles
- The alignment of until/price is to the right, which does not match the alignment of any other text, it should be aligned to the left as well.
- The border of the top information part end in a different spot than the border that split the information and the description.
- The spacing of the text is different from the spacing of the registrations.
- There is no header above the registrations that says 'registrations'.
- When cancelling a registration the 'No' button and the title in the dialog are not uppercased (whenever using Oswald the text should be uppercased).
- It is not possible to register again for an event that you have cancelled your registration for when the registration deadline has not passed yet (example at the time of writing: Thalympics).
- It would be better if the vertical padding of all parts if more than the horizontal padding. Preferably doubled so that we're not trying to put everything in 1cm2. In general everything could use a bit more space. Good example where this does not work is when there are no registrations and the 'There are no registrations yet.' text is shown.

### How to reproduce
Steps to reproduce the behaviour:
1. Go to an event detail view

### PR DESCRIPTION
**Pull request by LCKnol**
Closes #128 

### Summary

- Adds a "Registrations" header above the registrations in an event.
- Change "No" button to uppercase when cancelling for an event.
- Fix padding to the left and right of registrations to be equal to the rest of the event screen.

### CODE REVIEW

**Comment by DeD1rk** [comment]
So far all of the point in #128 have been dealt with, except:

- [ ] It is not possible to register again for an event that you have cancelled your registration for when the registration deadline has not passed yet (example at the time of writing: Thalympics).


There is also:
> The border of the top information part end in a different spot than the border that split the information and the description.

This is a material design 'middle divider' for medium emphasis grouping in related content, whereas the other one divides the screen in sections (map | info | description | registrations).

File: lib/ui/widgets/menu_drawer.dart [hunk 1]
```diff
@@ -84,13 +77,23 @@ class MenuDrawer extends StatelessWidget {
                         ),
                       ),
                     ),
+                    Positioned.fill(
+                      child: Material(
+                        color: Colors.transparent,
+                        child: InkWell(
+                          onTap: () {
+                            routerDelegate.push(TypedMaterialPage(
```
Comment (TippieCodes) [inline lib/ui/widgets/menu_drawer.dart:85]: Strange opening animation where fade in comes from below on andriod

File: pubspec.lock [hunk 2]
```diff
@@ -42,7 +42,7 @@ packages:
       name: async
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "2.8.2"
```
Comment (TippieCodes) [inline pubspec.lock:45]: This does not seem to belong in this PR

File: pubspec.lock [hunk 3]
```diff
@@ -526,7 +526,7 @@ packages:
       name: matcher
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "0.12.11"
+    version: "0.12.10"
```
Comment (TippieCodes) [inline pubspec.lock:529]: This does not seem to belong in this PR

File: pubspec.lock [hunk 4]
```diff
@@ -944,7 +944,7 @@ packages:
       name: test_api
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "0.4.3"
+    version: "0.4.2"
```
Comment (TippieCodes) [inline pubspec.lock:947]: This does not seem to belong in this PR

File: lib/ui/screens/event_screen.dart [hunk 5]
```diff
@@ -918,15 +953,13 @@ class _EventScreenState extends State<EventScreen> {
...
-            body: const Center(
-              child: CircularProgressIndicator(),
-            ),
+            body: const Center(child: CircularProgressIndicator()),
           );
         } else {
           final event = (state.result ?? widget.event)!;
           return Scaffold(
             appBar: ThaliaAppBar(
-              title: Text(event.title.toUpperCase()),
```
Comment (TippieCodes) [inline lib/ui/screens/event_screen.dart:929]: Personally I find it better to keep this as the name or maybe prefix it with EVENT:, but leave the name of the event in, this is useful when an event has a longer description or a lot of members and you forgot the name whilst reading the event information.

Comment (TippieCodes) [reply lib/ui/screens/event_screen.dart:929]: This has been decided with available technicie members to revert

**Review by TippieCodes — CHANGES_REQUESTED** [review]
- Strange opening animation where fade in comes from below on android
- (pubspec.lock) This does not seem to belong in this PR
- (event_screen.dart) Personally I find it better to keep this as the name or maybe prefix it with EVENT:, but leave the name of the event in, this is useful when an event has a longer description or a lot of members and you forgot the name whilst reading the event information.

**Comment by TippieCodes** [comment]
Above commit has been discussed and decided by anyone at the technicie meeting

File: lib/ui/widgets/menu_drawer.dart (hunk 1 above)
Comment (TippieCodes) [reply lib/ui/widgets/menu_drawer.dart:85]: Resolving this on this PR, this will be looked at after PR (#157) has been merged

**Review by TippieCodes — APPROVED** [review]

**Pull request merged**